File: 171.cpp
// 171. Excel Sheet Column Number

class Solution {
public:
    int titleToNumber(string s) {
        int res = 0 , l = 0;
        for (int i=s.length()-1;i>=0;i--) {
            res += pow (26,l)*(s[i]-'A'+1);
            l++;
        }
        return res;
    }
};

File: 171.py
class Solution:
    def titleToNumber(self, s: str) -> int:
        ret = 0
        eta = 1
        
        for i in range(len(s)):
            ret += ((ord(s[len(s)-i-1])-ord('A')) + 1) * eta
            eta *= 26
        
        return ret
        
